usng4j-impl/src/main/java/org/codice/usng4j/impl/CoordinateSystemTranslatorImpl.java

```diff
@@ -279,13 +291,49 @@ public UtmCoordinate toUtm(final DecimalDegreesCoordinate decimalDegreesCoordina
     return new UtmCoordinateImpl(zoneNumber, UTMEasting, UTMNorthing);
   }
 
-  /** {@inheritDoc} */
+  @Override
+  public UpsCoordinate toUps(final DecimalDegreesCoordinate decimalDegreesCoordinate) {
+    final UtmUpsCoordinate utmUpsCoordinate = toUtmUps(decimalDegreesCoordinate);
+    if (utmUpsCoordinate.isUTM()) {
+      throw new IllegalArgumentException(
+          utmUpsCoordinate + " is a UTM coordinate, please use toUtm or toUtmUps functions.");
+    }
+    return utmUpsCoordinate;
+  }
+
+  @Override
+  public UtmUpsCoordinate toUtmUps(final DecimalDegreesCoordinate decimalDegreesCoordinate) {
+    validateDecimalDegreeInput(decimalDegreesCoordinate);
+    final boolean northPole = decimalDegreesCoordinate.getLat() >= 0.0;
```

Comment:
❓ This boolean represents that the latitude is in the northern hemisphere, not at the north pole, right? If so, should rename to clarify that. Maybe `isNorthern` or similar.

Reply:
This used for calculating UPS and hence determines the poles.